libraries/AP_Compass/CompassCalibrator.cpp

```diff
@@ -1065,14 +1065,15 @@ bool CompassCalibrator::calculate_orientation(void)
  */
 bool CompassCalibrator::fix_radius(void)
 {
-    if (AP::gps().status() < AP_GPS::GPS_OK_FIX_2D) {
+    Location loc;
+    if (!AP::ahrs().get_location(loc)) {
```

Comment:
i think we need to accept either AHRS location or GPS location, so lack of healthy compass doesn't prevent compass calibration to fix the issue

Reply:
could also do AP::ahrs().get_origin()